src/PyDynamic/uncertainty/propagate_DFT.py

```diff
@@ -383,58 +383,75 @@ def GUM_iDFT(
 
-    Raises
-    ------
-    ValueError
-        If Nx is not smaller than dimension of UF - 2
     """
-    N = UF.shape[0] - 2
+    # (complex) input length
+    N_in = F.size // 2
 
+    # default output length, assumes even output length
+    N_out = Nx
+    N_out_default = UF.shape[0] - 2
     if Nx is None:
-        Nx = N
-    elif Nx > UF.shape[0] - 2:
-        raise ValueError(
-            f"GUM_iDFT: Nx if provided is expected to be smaller or equal to number "
-            f"of rows and columns of UF - 2, but UF is of shape {UF.shape} and Nx ="
-            f" {Nx}."
-        )
+        N_out = N_out_default
+
+    # calculate discrete angular frequency
+    beta = 2 * np.pi * np.arange(N_out) / N_out
+
+    # calculate inverse DFT
+    x = np.fft.irfft(F[:N_in] + 1j * F[N_in:], n=N_out)
 
-    beta = 2 * np.pi * np.arange(Nx) / N
+    # propagate uncertainty
+    if not isinstance(Cc, np.ndarray) or not isinstance(Cs, np.ndarray):
+        k = np.arange(N_in)
+        bk = np.outer(beta, k)
 
-    # calculate inverse DFT; Note: scaling factor 1/N is accounted for at the end
-    x = np.fft.irfft(F[: N // 2 + 1] + 1j * F[N // 2 + 1 :])[:Nx]
-    if not isinstance(Cc, np.ndarray):  # calculate sensitivities
-        Cc = np.zeros((Nx, N // 2 + 1))
-        Cc[:, 0] = 1.0
-        Cc[:, -1] = np.cos(np.pi * np.arange(Nx))
-        for k in range(1, N // 2):
-            Cc[:, k] = 2 * np.cos(k * beta)
+    # calculate sensitivities (scaling factor 1/N_out is accounted for at the end)
+    if not isinstance(Cc, np.ndarray):
+        Cc = np.cos(bk)
+        Cc = _adjust_sensitivity_matrix_iDFT(Cc, N_out, N_out_default)
 
     if not isinstance(Cs, np.ndarray):
-        Cs = np.zeros((Nx, N // 2 + 1))
-        Cs[:, 0] = 0.0
-        Cs[:, -1] = -np.sin(np.pi * np.arange(Nx))
-        for k in range(1, N // 2):
-            Cs[:, k] = -2 * np.sin(k * beta)
+        Cs = -np.sin(bk)
+        Cs = _adjust_sensitivity_matrix_iDFT(Cs, N_out, N_out_default)
 
     # calculate blocks of uncertainty matrix
     if len(UF.shape) == 2:
-        RR = UF[: N // 2 + 1, : N // 2 + 1]
-        RI = UF[: N // 2 + 1, N // 2 + 1 :]
-        II = UF[N // 2 + 1 :, N // 2 + 1 :]
+        RR = UF[:N_in, :N_in]
+        RI = UF[:N_in, N_in:]
+        II = UF[N_in:, N_in:]
         # propagate uncertainties
         Ux = np.dot(Cc, np.dot(RR, Cc.T))
-        Ux = Ux + 2 * np.dot(Cc, np.dot(RI, Cs.T))
+        Ux += 2 * np.dot(Cc, np.dot(RI, Cs.T))
         Ux += np.dot(Cs, np.dot(II, Cs.T))
     else:
-        RR = UF[: N // 2 + 1]
-        II = UF[N // 2 + 1 :]
+        RR = UF[:N_in]
+        II = UF[N_in:]
         Ux = np.dot(Cc, _prod(RR, Cc.T)) + np.dot(Cs, _prod(II, Cs.T))
 
     if returnC:
-        return x, Ux / N**2, {"Cc": Cc, "Cs": Cs}
+        return x, Ux / N_out**2, {"Cc": Cc, "Cs": Cs}
     else:
-        return x, Ux / N**2
+        return x, Ux / N_out**2
+
+
+def _adjust_sensitivity_matrix_iDFT(C, N_out, N_out_default):
```

Comment:
To what? Can we reasonably change or extend the name to give more information about what is happening?

Reply:
Resolved by 24adf2f.